.github/workflows/collector-generate-and-update.yml

```diff
@@ -0,0 +1,71 @@
+name: Update OTel Collector Dependencies and Generate OTel Agent
+
+on:
+  schedule:
+    - cron: '0 12 * * 3'
+  workflow_dispatch:
+
+jobs:
+  update-and-generate:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set date
+        id: date
+        run: echo "date=$(date +'%Y-%m-%d')" >> $GITHUB_OUTPUT
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12.6'
```

Comment:
Could you use something like [this](https://github.com/DataDog/datadog-agent/blob/04a7d9aafdf9815eb7901d487b61f62bf7a96b04/.github/workflows/label-analysis.yml#L29-L34) ? With the hash of the proper version